packages/framework/src/Model/Transport/TransportFacade.php

```diff
@@ -200,44 +206,64 @@ public function getAllIncludingDeleted()
+     * @param int $domainId
      * @return \Shopsys\FrameworkBundle\Component\Money\Money[]
      */
-    public function getTransportPricesWithVatIndexedByTransportId(Currency $currency): array
+    public function getTransportPricesWithVatByCurrencyAndDomainIdIndexedByTransportId(Currency $currency, int $domainId): array
     {
         $transportPricesWithVatByTransportId = [];
         $transports = $this->getAllIncludingDeleted();
         foreach ($transports as $transport) {
-            $transportPrice = $this->transportPriceCalculation->calculateIndependentPrice($transport, $currency);
+            $transportPrice = $this->transportPriceCalculation->calculateIndependentPrice($transport, $currency, $domainId);
             $transportPricesWithVatByTransportId[$transport->getId()] = $transportPrice->getPriceWithVat();
         }
 
         return $transportPricesWithVatByTransportId;
     }
 
     /**
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Currency\Currency $currency
+     * @param int $domainId
      * @return string[]
      */
-    public function getTransportVatPercentsIndexedByTransportId()
+    public function getTransportVatPercentsByCurrencyAndDomainIdIndexedByTransportId(Currency $currency, int $domainId): array
     {
         $transportVatPercentsByTransportId = [];
         $transports = $this->getAllIncludingDeleted();
         foreach ($transports as $transport) {
-            $transportVatPercentsByTransportId[$transport->getId()] = $transport->getVat()->getPercent();
+            $transportVatPercentsByTransportId[$transport->getId()] = $transport->getVatByCurrencyAndDomainId($currency, $domainId)->getPercent();
         }
 
         return $transportVatPercentsByTransportId;
     }
 
     /**
      * @param \Shopsys\FrameworkBundle\Model\Transport\Transport $transport
-     * @return \Shopsys\FrameworkBundle\Model\Pricing\Price[]
+     * @return \Shopsys\FrameworkBundle\Model\Pricing\Price[][]
      */
-    public function getIndependentBasePricesIndexedByCurrencyId(Transport $transport)
+    public function getIndependentBasePricesByDomainIdIndexedByDomainIdAndCurrencyId(Transport $transport): array
     {
         $prices = [];
         foreach ($transport->getPrices() as $transportInputPrice) {
+            $domainId = $transportInputPrice->getDomainId();
             $currency = $transportInputPrice->getCurrency();
-            $prices[$currency->getId()] = $this->transportPriceCalculation->calculateIndependentPrice($transport, $currency);
+            $prices[$domainId][$currency->getId()] = $this->transportPriceCalculation->calculateIndependentPrice($transport, $currency, $domainId);
+        }
+
+        return $prices;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Transport\Transport|null $payment
+     * @return \Shopsys\FrameworkBundle\Model\Pricing\Price[]
+     */
+    public function getPricesIndexedByDomainId(?Transport $payment): array
```

Comment:
Transport can be nullable and is named $payment, please fix

Reply:
Transport is nullable, or not? I fixed name

Reply:
I do not think it should be nullable - you added nullability why?

Reply:
Ok, when I want create new transport - I want array of prices, but at this moment is Transport null yet. And I don't want create if-esle in template rendering like a previosly solution with currencies.

Reply:
Well before you did in FormType this:

`$payment !== null ? $this->paymentFacade->getIndependentBasePricesIndexedByCurrencyId($payment) : []`

Now you moved this logic to `calculateIndependentPrice` and added nullability to first parameter... I prefer previous solution

Reply:
Hmm, I don't like previos solution. In my solution I have one condition - in TransportPriceCalculation - when is transport object null -> return Price::zero. In previous solution I had condition in facade and in form template. Plus form template had different behavior, because when array item not existed, form did't print element - in my solution is in element zero price and it is better for me.

Reply:
But I changed it a little bit :smile: